src/services/fetchers/PolygonIOCryptoPriceFetcher.ts

```diff
@@ -13,17 +13,22 @@ class PolygonIOCryptoPriceFetcher {
 
   static fetcherSource = '';
 
-  async apply({fsym, tsym}: {fsym: string; tsym: string}, timestamp: number): Promise<number> {
+  async apply(params: {fsym: string; tsym: string}, options: FeedFetcherOptions): Promise<number> {
+    const {fsym, tsym} = params;
+    const {base: feedBase, quote: feedQuote, timestamp} = options;
+
+    if (!timestamp || timestamp <= 0) throw new Error(`invalid timestamp value: ${timestamp}`);
+
     const price = await this.polygonIOCryptoPriceService.getLatestPrice({fsym, tsym}, timestamp);
 
     if (price !== null) {
       const payload: PriceDataPayload = {
         fetcher: FetcherName.POLYGON_IO_CRYPTO_PRICE,
         value: price.toString(),
         valueType: PriceValueType.Price,
-        timestamp: this.timeService.apply(),
-        feedBase: fsym,
-        feedQuote: tsym,
+        timestamp,
```

Comment:
ok this was probably a bug, and we should use timestamp from argument if `getLatestPrice` is fetching price base on `timestamp`